component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -292,14 +293,15 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         }
         String userToken = request.getParameter(EmailOTPAuthenticatorConstants.CODE);
         String contextToken = (String) context.getProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN);
-        if (userToken.equals(contextToken)) {
+        long genTime = (long)context.getProperty(EmailOTPAuthenticatorConstants.OTP_GENERATED_TIME);
+        if (userToken.equals(contextToken)  && System.currentTimeMillis()<=genTime+new Long(new EmailOTPAuthenticator().getExpireTime(context))) {
```

Comment:
Not formatted

Reply:
No need of instantiating a new object of EmailOTPAuthenticator. You can directly use the private method

Reply:
Better to add another method like 'isExpired' and handle the expiration logic there so that it is more readable